virttest/utils_net.py

```diff
@@ -3118,55 +3200,31 @@ def update_mac_ip_address(vm, params, timeout=None):
     :param vm: VM object
     :param params: Dictionary with the test parameters.
     """
-    network_query = params.get("network_query", "ifconfig")
-    restart_network = params.get("restart_network", "service network restart")
-    filter_list = []
-    mac_ip_filter = params.get("mac_ip_filter")
-    if mac_ip_filter:
-        filter_list.append(mac_ip_filter)
-    else:
-        # Provide a list of patterns for different linux version
-        filter_list.extend([(r"HWaddr (.\w+:\w+:\w+:\w+:\w+:\w+)\s+?"
-                             "inet addr:(.\d+\.\d+\.\d+\.\d+)"),
-                            (r"inet (.\d+.\d+.\d+.\d+).*?"
-                             "ether (.\w+:\w+:\w+:\w+:\w+:\w+)")])
-    if timeout is None:
-        timeout = int(params.get("login_timeout"))
     session = vm.wait_for_serial_login(timeout=360)
-    end_time = time.time() + timeout
-    macs_ips = []
-    num = 0
-    while time.time() < end_time:
-        try:
-            if num % 3 == 0 and num != 0:
-                # Ignore any errors here for further operation
-                session.cmd(restart_network, ignore_all_errors=True)
-            output = session.cmd_status_output(network_query)[1]
-            for mac_ip_filter in filter_list:
-                macs_ips = re.findall(mac_ip_filter, output, re.S)
-                if macs_ips:
-                    break
-            # Get nics number
-        except Exception, err:
-            logging.error(err)
-        nics = params.get("nics")
-        nic_minimum = len(re.split(r"\s+", nics.strip()))
-        if len(macs_ips) == nic_minimum:
-            break
-        num += 1
-        time.sleep(5)
-    if len(macs_ips) < nic_minimum:
-        logging.error("Not all nics get ip address")
-
-    for (_ip, mac) in macs_ips:
-        vlan = macs_ips.index((_ip, mac))
-        # _ip, mac are in different sequence in Fedora and RHEL guest.
-        if re.match(".\d+\.\d+\.\d+\.\d+", mac):
-            _ip, mac = mac, _ip
+    maps = get_mac_ip_maps(session)
+    if len(maps) < len(vm.virtnet):
+        logging.warn("Not all nic get IP address, restart networking")
+        default_cmd = "service network restart"
+        restart_network = params.get("restart_network", default_cmd)
+        session.cmd(restart_network, timeout=240)
+        maps = get_mac_ip_maps(session)
```

Comment:
IMHO, Now we're checking code on paper. we don't need to worry about the things not happen.
repetition can be simply used to fix this kind of problem. but it's really bad solution